src/command_modules/azure-cli-interactive/azclishell/__main__.py

```diff
@@ -38,9 +35,18 @@ def main(style=None):
     CONFIG.load(os.path.join(azure_folder, 'az.json'))
     SESSION.load(os.path.join(azure_folder, 'az.sess'), max_age=3600)
 
-    config = SHELL_CONFIGURATION
+    config = azclishell.configuration.CONFIGURATION
     shell_config_dir = azclishell.configuration.get_config_dir
 
+    try:
+        commands = GatherCommands()
+        az_completer = AzCompleter(commands)
+        cache_load = True
+    except IOError:  # if there is no cache
```

Comment:
When it regenerates the command table in the background, that one will be dumped. So each time the cache is refreshed.